test_integration/lib/test/realtime/realtime_history_with_auth_callback_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:ably_flutter/ably_flutter.dart';
+import 'package:ably_flutter_integration_test/app_provisioning.dart';
+import 'package:ably_flutter_integration_test/config/test_constants.dart';
+import 'package:ably_flutter_integration_test/factory/reporter.dart';
+import 'package:ably_flutter_integration_test/utils/realtime.dart';
+
+Future<Map<String, dynamic>> testRealtimeHistoryWithAuthCallback({
+  required Reporter reporter,
+  Map<String, dynamic>? payload,
+}) async {
+  var authCallbackInvoked = false;
+
+  final realtime = Realtime(
+    options: ClientOptions(
+      clientId: 'someClientId',
+      logLevel: LogLevel.verbose,
+      authCallback: (params) async {
+        authCallbackInvoked = true;
+        return TokenRequest.fromMap(
+          await AppProvisioning().getTokenRequest(),
+        );
+      },
+    ),
+  );
+
+  final channel = realtime.channels.get('test');
+  await publishMessages(channel);
+
+  await channel.history();
+  await Future<void>.delayed(TestConstants.publishToHistoryDelay);
+
+  return {
+    'handle': await realtime.handle,
+    'authCallbackInvoked': authCallbackInvoked,
```

Comment:
Instead of checking if the auth callback was invoked maybe we should check if the `channel.history()` has returned the correct result? Or maybe both this and the auth callback? :thinking: I'm wondering what exactly do we want to test here :wink:

Reply:
There's an entirely separate suite of tests checking correctness of history in [rest_history_test.dart](https://github.com/ably/ably-flutter/blob/main/test_integration/lib/test/rest/rest_history_test.dart). These tests are only supposed to check if `authCallback` is executed when specific rest/realtime methods are called. The structure of tests like this already exists in a project, for example in [rest_publish_with_auth_callback.test](https://github.com/ably/ably-flutter/blob/main/test_integration/lib/test/rest/rest_publish_with_auth_callback_test.dart), I only added a missing one for history.